### ISSUE TITLE
Add "new in version" tag to documentation

### ISSUE THREAD

**Issue by c2thorn** [hashicorp/terraform-provider-google#13538]
Suggestion to add some sort of systematic version introduction tag to fields in our documentation. Can see an argument for added clutter, but it should be useful information.

Suggested by a user who said it would have helped them: https://github.com/hashicorp/terraform-provider-google/issues/13398#issuecomment-1398752774

An example in how python does this: https://docs.python.org/3/library/os.path.html#os.path.commonpath

**Comment by rileykarson** [comment]
I think HashiCorp's intended model here is for users to pick old versions with the Registry version selector. However, I don't know how widely that's used by end users in practice. It'd be interesting if they solved this on their end- using structured metadata instead of raw markdown for docs pages could solve the issue ecosystem-wide, at the cost of worse ergonomics when writing docs pages (since each field would likely still be MD, and there's likely need to be a "header" entry where users throw a bunch of markdown.

**Comment by Rycieos** [comment]
> I think HashiCorp's intended model here is for users to pick old versions with the Registry version selector. However, I don't know how widely that's used by end users in practice.

Not at all, and that is counter-intuitive. When reading the documentation, my first though is not that I might be running an out of date version. I had no idea that the version I was running was even out of date anyway. And if I did pick the version I had currently installed on the docs site, I would never have seen that a new flag was added, so I would not have wanted to update. But if was on the current version docs, and saw the field with a note "version added: 4.48.0", I would have considered what version I was currently on, and then choose to upgrade.
